### PR TITLE
Fix count tokens to include model params

### PR DESCRIPTION
**Pull request by hsubox76**
Count tokens request should include any params set on the model parent class, including systemInstruction, tools, cachedContent.

Updated samples to use this.

generationConfig, toolConfig, and safetySettings don't seem like they would affect tokenCount but might as well pass them through, as they are part of a complete GenerateContentRequest and may affect tokenCount in the future.

### CODE REVIEW

File: packages/main/src/requests/request-helpers.ts [hunk 1]
```diff
@@ -111,9 +113,30 @@ function assignRoleToPartsAndValidateSendMessageRequest(
...
+  //     model: modelParams?.model,
+  //     generationConfig: modelParams?.generationConfig,
+  //     safetySettings: modelParams?.safetySettings,
+  //     tools: modelParams?.tools,
+  //     toolConfig: modelParams?.toolConfig,
+  //     systemInstruction: modelParams?.systemInstruction,
+  //     cachedContent: modelParams?.cachedContent?.name,
+  //     contents: [],
+  //   },
+  // };
```
Comment (DellaBitta) [inline packages/main/src/requests/request-helpers.ts:139]: Should this be removed?

File: packages/main/src/requests/request-helpers.test.ts [hunk 2]
```diff
@@ -172,4 +177,104 @@ describe("request formatting methods", () => {
       });
     });
   });
+  describe("formatCountTokensInput", () => {
+    it("formats a text string into a count request", () => {
+      const result = formatCountTokensInput("some text content", {
+        model: "gemini-1.5-flash",
+      });
+      // expect(result.generateContentRequest.model).to.equal("gemini-1.5-flash");
+      // expect(result.generateContentRequest.contents).to.equal("gemini-1.5-flash");
```
Comment (DellaBitta) [inline packages/main/src/requests/request-helpers.test.ts:186]: Remove these?

**Review by DellaBitta — CHANGES_REQUESTED** [review]

File: packages/main/src/requests/request-helpers.ts (hunk 1 above)
Comment (hsubox76) [reply packages/main/src/requests/request-helpers.ts]: Yep, thanks!

File: packages/main/src/requests/request-helpers.test.ts (hunk 2 above)
Comment (hsubox76) [reply packages/main/src/requests/request-helpers.test.ts]: Done.

**Review by DellaBitta — APPROVED** [review]

**Pull request merged**